### PR TITLE
EVG-5700 clone over HTTPS in fetch, make auth errors non-interactive

### PR DESCRIPTION
**Pull request by john-m-liu**

### CODE REVIEW

File: operations/fetch.go [hunk 1]
```diff
@@ -115,7 +123,7 @@ func Fetch() cli.Command {
 			}
 
 			if doFetchSource {
-				if err = fetchSource(ctx, ac, rc, client, wd, taskID, noPatch); err != nil {
+				if err = fetchSource(ctx, ac, rc, client, wd, taskID, c.String(tokenFlagName), noPatch); err != nil {
```
Comment (tychoish) [inline operations/fetch.go]: move the c.String call up to line 108

File: operations/fetch.go [hunk 2]
```diff
@@ -14,6 +14,8 @@ import (
 	"strings"
 	"sync"
 
+	"github.com/evergreen-ci/evergreen/thirdparty"
+
```
Comment (tychoish) [inline operations/fetch.go]: why blank line?

File: command/git.go [hunk 3]
```diff
@@ -147,7 +147,7 @@ func (opts cloneOpts) buildHTTPCloneCommand() ([]string, error) {
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to parse URL from location")
 	}
-	clone := fmt.Sprintf("git clone https://%s@%s/%s/%s.git '%s'", opts.token, urlLocation.Host, opts.owner, opts.repo, opts.dir)
+	clone := fmt.Sprintf("git clone https://%s:x-oauth-basic@%s/%s/%s.git '%s'", opts.token, urlLocation.Host, opts.owner, opts.repo, opts.dir)
```
Comment (tychoish) [inline command/git.go]: should we make a function to produce these strings as a method on opts or as a function?

File: operations/fetch_test.go [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+	for name, opts := range passingCases {
+		t.Run(name, func(t *testing.T) {
+			runCloneTest(t, opts, true)
+		})
+	}
+	for name, opts := range failingCases {
+		t.Run(name, func(t *testing.T) {
+			runCloneTest(t, opts, false)
+		})
+	}
```
Comment (tychoish) [inline operations/fetch_test.go]: I would have put the test cases in a struct with name and weather it is passing or failing, and just have the runCloneTest in the same t.Run body.

Comment (john-m-liu) [reply operations/fetch_test.go]: fixed

**Pull request merged**